src/gpodder/extensions.py

```diff
@@ -380,16 +391,31 @@ def _find_extensions(self):
             user_extensions = os.path.join(gpodder.home, 'Extensions', '*.py')
             self.filenames = glob.glob(builtins) + glob.glob(user_extensions)
 
-        # Let user extensions override built-in extensions of the same name
+        # Let user extensions override built-in extensions of the same name.
+        # This inherently happens because we search the user extensions folder second,
+        # and the entries are put in the extensions dict by their name field.
         for filename in self.filenames:
             if not filename or not os.path.exists(filename):
                 logger.info('Skipping non-existing file: %s', filename)
                 continue
 
             name, _ = os.path.splitext(os.path.basename(filename))
-            extensions[name] = filename
 
-        return sorted(extensions.items())
+            # get ordering prefix
+            priority = self._get_prefix(name)
+
+            # strip ordering prefix from name, if present
+            name = re.sub(r'^[0-9]*_', '', name)
+            try:
+                if extensions[name] is not None:
+                    logger.info("extension at %s will be ignored in favor of %s", extensions[name][1], filename)
+            except:
+                None
+            extensions[name] = (priority, filename)
```

Comment:
Untested, but maybe you could get rid of _get_prefix() and write it like that.
I prefer playing with the prefix in a single place. I also prefer to use `key in dict` to avoid the `try/except`

```suggestion
            priority = 99
            m = re.fullmatch(r'^([0-9]*)_(.+)', name)
            if m:
              if m.group(1):
                # get ordering prefix
                priority = int(m.group(1))
              # strip ordering prefix from name (or leading _)
              name = m.group(2)
              if name in extensions:
                    logger.info("extension at %s will be ignored in favor of %s", extensions[name][1], filename)
            extensions[name] = (priority, filename)
```

Reply:
This seems like a bit of pythonism I'm not familiar with, but I'll parse through it and see if it does the same thing, sure.

Reply:
After parsing through it and trying it out, yeah it seems reasonable enough. I originally thought there would be other places _get_prefix() would be used, but that was before we were saving it as its own value.